### ISSUE TITLE
Enum type is "expanded" which cause unexpected errors

### ISSUE THREAD

**Issue by moi15moi** [python/mypy#21268]
**Bug Report**

Since the version 1.20.0, I get this error, mypy seems to expand the enum type which cause unexpected errors.
With the version 1.19.1, i wasn't getting the error, so I guess it's a regression.

In my example, it consider `RoundingMethod.ROUND` to have the type `Callable[[Fraction], int]` instead of the type `RoundingMethod`.

**To Reproduce**

Use this gist https://mypy-play.net/?mypy=latest&python=3.12&gist=d0631665634be1d6c9fdb4b32bc41e4f or this:
```python
from collections.abc import Callable
from enum import Enum
from fractions import Fraction
from math import ceil, floor

RoundingCallType = Callable[[Fraction], int]

def floor_method(number: Fraction) -> int:
    return floor(number)

def round_method(number: Fraction) -> int:
    if number >= 0:
        return floor(number + Fraction(1, 2))
    else:
        return ceil(number - Fraction(1, 2))

class RoundingMethod(Enum):
    FLOOR = floor_method
    ROUND = round_method

    def __call__(self, number: Fraction) -> int:
        method: RoundingCallType = self.value
        return method(number)


def round(rounding_method: RoundingMethod, a_number: Fraction) -> int:
    return rounding_method(a_number)

def main() -> None:
    a_number = Fraction(6, 5)
    rounded_number = round(RoundingMethod.ROUND, a_number)
    print(f"The rounded_number is {rounded_number}")

if __name__ == "__main__":
    main()
```

**Expected Behavior**

mypy shouldn't report a error, so it should report:
```
Success: no issues found in 1 source file
```

**Actual Behavior**

```
test.py:31: error: Argument 1 to "round" has incompatible type "Callable[[Fraction], int]"; expected "RoundingMethod"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.20.0
- Mypy command-line flags: ``mypy test.py``
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.12

**Comment by A5rocks** [comment]
This is because those callables are not enum members. See https://github.com/python/mypy/issues/20739
